Update the hello_world tutorial

rajarshimaitra opened the pull request:
Finishes up https://github.com/bitcoindevkit/bitcoindevkit.org/pull/93.

Couldn't get the link reference list thingy working for `SqliteDatabase` link so made an in place link.

mplsgrant committed:
Update the hello_world tutorial

Co-authored-by: mpls <mpls@posteo.net>

vladimirfomene commented on docs/tutorials/hello-world.md:
```diff
@@ -180,17 +160,14 @@ println!(
-We briefly mentioned before that for our example we used the `MemoryDatabase`, but that it could also be swapped for a different one: this is one example of the *modularity* of BDK. By default, some database
-types are implemented in the library, namely (as of now) the `MemoryDatabase` which only keeps data in RAM and the [sled][sled] database that can store data on a filesystem. But since the `Database` trait is public,
-users of the library can also implement different database types more suitable for their use-case.
+We briefly mentioned before that for our example we used the `MemoryDatabase`, but that it could also be swapped for a different one: this is one example of the *modularity* of BDK. By default, some database types are already implemented in the library, namely the `MemoryDatabase` which only keeps data in RAM, the [sled][sled] database that can store data on a filesystem, and the [`SqliteDatabase ...
```
@rajarshimaitra I was able to get the reference link working on my end using `[\`SqliteDatabase\`][SqliteDatabase]` in the text and `[SqliteDatabase]: https://docs.rs/bdk/latest/bdk/database/struct.SqliteDatabase.html` in the reference section at the end.

rajarshimaitra replied:
Thanks, yes I updated the link and this time it seems to be working.. Can you check again?

vladimirfomene commented on docs/tutorials/hello-world.md:
````diff
@@ -170,8 +147,11 @@ And then broadcast it:
 ```rust
 // ...
 
+use bdk::blockchain::Blockchain
````
Nits: Missing semicolon

rajarshimaitra replied:
Done..

vladimirfomene commented on docs/tutorials/hello-world.md:
````diff
@@ -75,13 +68,11 @@ Building and running this code will print out:
+We can update our code to look something like this:
 
 ```rust
 use bdk::{
````
This import does not have `wallet::AddressIndex` like the import section above.

rajarshimaitra replied:
Good catch.. Done..

vladimirfomene reviewed (commented):
I pulled the branch and built the wallet while following the tutorial. Tested ACK

rajarshimaitra committed:
Remove previous Hello World blog post

rajarshimaitra commented:
Thanks @vladimirfomene for the review.. I updated as suggest, rebased on master, and removed the old Hello World blog post.. Which was a previous duplicate of this tutorial.

notmandatory commented:
Looks like this one is ready to go but needs a final rebase. Sorry for the long delay!

notmandatory committed:
Merge branch 'master' into update-helloworld

notmandatory reviewed (approved):
utACK 084e24e5a37beed913461c59693642747939f3cb
